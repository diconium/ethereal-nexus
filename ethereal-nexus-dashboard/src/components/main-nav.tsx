import Link from "next/link";

import { cn } from "@/lib/utils";

export function MainNav({
  className = "",
  ...props
}: React.HTMLAttributes<HTMLElement>) {
  return (
    <nav
      className={cn("flex items-center space-x-4 lg:space-x-6", className)}
      {...props}
    >
      <Link
        href="/"
        className="text-sm font-medium transition-colors hover:text-primary"
      >
        Home
      </Link>
      <Link
        href="/components"
        className="text-sm font-medium text-muted-foreground transition-colors hover:text-primary"
      >
        Components
      </Link>
      <Link
        href="/users"
        className="text-sm font-medium text-muted-foreground transition-colors hover:text-primary"
      >
        Users
      </Link>
      <Link
        href="/api-doc"
        className="text-sm font-medium text-muted-foreground transition-colors hover:text-primary"
      >
        API Documentation
      </Link>
    </nav>
  );
}
