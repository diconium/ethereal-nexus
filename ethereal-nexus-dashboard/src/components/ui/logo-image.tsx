'use client'
import { useEffect, useState } from 'react';
import { useTheme } from 'next-themes';
import Image from 'next/image';
export const LogoImage = () => {
  const imageSizePixels = 267;
  const {theme} = useTheme();
  const logoWithSystemColor = typeof window !== 'undefined' &&
  window.matchMedia('(prefers-color-scheme: dark)').matches ? '/ethereal-nexus-logo-dark-mode.jpg' : '/ethereal-nexus-logo-light-mode.jpg';
  const [image, setImage] = useState(logoWithSystemColor);

  useEffect(() => {
    theme === 'system' ?  setImage(logoWithSystemColor) : setImage(`/ethereal-nexus-logo-${theme}-mode.jpg`);
  }, [theme]);

  return <Image data-testid={`ethereal-logo-${theme}-image`} src={image} alt="Ethereal Nexus Logo" width={imageSizePixels} height={imageSizePixels/9}/>;
};

export default LogoImage;
