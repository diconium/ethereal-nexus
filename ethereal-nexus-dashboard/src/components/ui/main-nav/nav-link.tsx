'use client'
import Link from "next/link";
import { cva, type VariantProps } from 'class-variance-authority';
import { ComponentProps } from 'react';
import { usePathname } from 'next/navigation';
import type { UrlObject } from "url";

const link = cva(["font-medium", "transition-colors"], {
  variants: {
    selected: {
      false: [''],
    },

  },
  defaultVariants: {
    selected: false,
  },
});

export interface NavLinkProps
  extends ComponentProps<typeof Link>,
    VariantProps<typeof link> {}

function isSelected(pathname: string, href: string | UrlObject) {
  if(pathname === href) {
    return true
  }

  return href !== '/' && pathname.startsWith(href.toString());
}

export function NavLink({href, className, children, target, rel}: NavLinkProps) {
  const pathname = usePathname()
  const selected = isSelected(pathname, href);

  const activeNavLink = selected ? 'bg-orange-600 rounded-full text-white py-4 px-8 flex justify-center' : '';
  const combinedClassName = `${activeNavLink} ${className}`;
  return (
      <Link
        href={href}
        target={target}
        rel={rel}
        className={link({ selected, className: combinedClassName })}
      >
        {children}
      </Link>
  );
}
