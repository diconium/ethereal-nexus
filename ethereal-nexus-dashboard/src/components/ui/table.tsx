import * as React from "react"

import {cn} from "@/lib/utils"
import {boolean} from "zod";

const Table = React.forwardRef<
    HTMLTableElement,
    React.HTMLAttributes<HTMLTableElement>
>(({className, ...props}, ref) => (
  <div className={`w-full overflow-auto border border-gray-300 rounded-lg transition-colors`}>
    <table
      ref={ref}
      className={cn('w-full caption-bottom text-base', className)}
            {...props}
        />
    </div>
))
Table.displayName = "Table"

const TableHeader = React.forwardRef<
    HTMLTableSectionElement,
    React.HTMLAttributes<HTMLTableSectionElement>
>(({className, ...props}, ref) => (
    <thead ref={ref} className={className} {...props} />
))
TableHeader.displayName = "TableHeader"

const TableBody = React.forwardRef<
    HTMLTableSectionElement,
    React.HTMLAttributes<HTMLTableSectionElement>
>(({className, ...props}, ref) => (
    <tbody
        ref={ref}
        className={cn("[&_tr:last-child]:border-0 dark:bg-transparent bg-accent", className)}
        {...props}
    />
))
TableBody.displayName = "TableBody"

const TableFooter = React.forwardRef<
    HTMLTableSectionElement,
    React.HTMLAttributes<HTMLTableSectionElement>
>(({className, ...props}, ref) => (
    <tfoot
        ref={ref}
        className={cn("bg-primary font-medium text-primary-foreground", className)}
        {...props}
    />
))
TableFooter.displayName = "TableFooter"

const TableRow = React.forwardRef<
    HTMLTableRowElement,
    React.HTMLAttributes<HTMLTableRowElement> & { highlightSelected?: boolean }
>(({className, highlightSelected = true, ...props}, ref) => (
    <tr
        ref={ref}
        className={cn(
            "transition-colors border",
            className
        )}
        {...props}
    />
))
TableRow.displayName = "TableRow"

const TableHead = React.forwardRef<
    HTMLTableCellElement,
    React.ThHTMLAttributes<HTMLTableCellElement>
>(({className, ...props}, ref) => (
    <th
        ref={ref}
        className={cn(
            "px-3 py-2 text-left align-middle font-medium [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]",
            className
        )}
        {...props}
    />
))
TableHead.displayName = "TableHead"

interface TableCellProps extends React.TdHTMLAttributes<HTMLTableCellElement> {
  columnIndex?: number;
}

const TableCell = React.forwardRef<
    HTMLTableCellElement,
    TableCellProps

>(({className, columnIndex, ...props}, ref) => (
    <td
        ref={ref}
        className={cn(
            "px-3 py-2 align-middle [&:has([role=checkbox])]:pr-0 [&>[role=checkbox]]:translate-y-[2px]",
            columnIndex === 0 && "w-1/4",
            columnIndex === 1 && "w-1/3",
            className
        )}
        {...props}
    />
))
TableCell.displayName = "TableCell"

const TableCaption = React.forwardRef<
    HTMLTableCaptionElement,
    React.HTMLAttributes<HTMLTableCaptionElement>
>(({className, ...props}, ref) => (
    <caption
        ref={ref}
        className={cn("mt-4 text-sm text-muted-foreground", className)}
        {...props}
    />
))
TableCaption.displayName = "TableCaption"

export {
    Table,
    TableHeader,
    TableBody,
    TableFooter,
    TableHead,
    TableRow,
    TableCell,
    TableCaption,
}
