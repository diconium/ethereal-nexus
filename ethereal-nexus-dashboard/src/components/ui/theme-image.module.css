.imgDark {
    display: none;
}

@media (prefers-color-scheme: dark) {
    .imgLight {
        display: none;
    }

    .imgDark {
        display: unset;
    }
}
