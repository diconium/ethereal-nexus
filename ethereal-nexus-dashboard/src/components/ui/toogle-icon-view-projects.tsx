"use client"

import React from 'react';
import CardViewDisabledIcon from '@/components/ui/icons/CardViewDisabledIcon';
import ListViewIcon from '@/components/ui/icons/ListViewIcon';
import ListViewDisabledIcon from '@/components/ui/icons/ListViewDisabledIcon';
import CardViewIcon from '@/components/ui/icons/CardViewIcon';
import { useViewMode } from '@/components/components/projects/ProjectsViewProvider';

export function ToogleIconViewProjects() {
  const { viewMode, toggleViewMode } = useViewMode();
  const isListView = viewMode === 'list';

  return (
    <div className="flex items-center ml-1">
      <button className="mr-1" onClick={toggleViewMode}>
        {!isListView ? (
          <CardViewIcon width={30} height={30} />
        ) : (
          <CardViewDisabledIcon width={30} height={30} />
        )}
      </button>
      <button onClick={toggleViewMode}>
        {isListView ? (
          <ListViewIcon width={30} height={30} />
        ) : (
          <ListViewDisabledIcon width={30} height={30} />
        )}
      </button>
    </div>
  );
}

