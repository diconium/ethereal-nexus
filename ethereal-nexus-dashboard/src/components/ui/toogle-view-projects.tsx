"use client"

import CardViewDisabledIcon from '@/components/ui/icons/CardViewDisabledIcon';
import ListViewIcon from '@/components/ui/icons/ListViewIcon';
import React, { useState } from 'react';
import ListViewDisabledIcon from '@/components/ui/icons/ListViewDisabledIcon';
import CardViewIcon from '@/components/ui/icons/CardViewIcon';

export function ToogleViewProjects() {
  const [viewMode, setViewMode] = useState('list'); // Initial view mode

  const toggleViewMode = () => {
    setViewMode(prevMode => (prevMode === 'card' ? 'list' : 'card'));
  };

  const isCardView = viewMode === 'card';

  return (
    <div className="flex items-center ml-1">
      <button className="mr-1" onClick={toggleViewMode}>
        {isCardView ? (
          <ListViewDisabledIcon width={30} height={30} />
        ) : (
          <CardViewDisabledIcon width={30} height={30} />
        )}
      </button>
      <button onClick={toggleViewMode}>
        {isCardView ? (
          <CardViewIcon width={30} height={30} />
        ) : (
          <ListViewIcon width={30} height={30} />
        )}
      </button>
    </div>
  );
}

