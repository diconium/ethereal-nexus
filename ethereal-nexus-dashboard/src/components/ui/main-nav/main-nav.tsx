import { cn } from "@/lib/utils";
import { NavLink } from './nav-link';

export function MainNav({
  className = "",
  ...props
}: React.HTMLAttributes<HTMLElement>) {
  return (
    <nav
      className={cn("flex items-center space-x-4 lg:space-x-6", className)}
      {...props}
    >
      <NavLink
        href="/"
      >
        Home
      </NavLink>
      <NavLink
        href="/projects"
      >
        Projects
      </NavLink>
      <NavLink
        href="/components"
      >
        Components
      </NavLink>
      <NavLink
        href="/users"
      >
        Users
      </NavLink>
      <NavLink
        href="/api-doc"
      >
        API Documentation
      </NavLink>
    </nav>
  );
}
