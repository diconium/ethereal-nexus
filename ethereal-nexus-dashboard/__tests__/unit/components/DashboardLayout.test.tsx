import React from 'react';
import { RenderResult, render, screen } from '@testing-library/react';
import DashboardLayout from '@/components/layout';
import LogoImage from '@/components/ui/logo-image';
import { MainNav } from '@/components/ui/main-nav/main-nav';
import ThemePicker from '@/components/theme-picker';
import { UserNav } from '@/components/user/user-nav';
import { Toaster } from '@/components/ui/toaster';


jest.mock('next-auth', () => ({
  signIn: jest.fn(),
}));

jest.mock('@/components/ui/logo-image', () => ({
  __esModule: true,
  default: jest.fn(),
}));

jest.mock('@/components/ui/main-nav/main-nav', () => ({
  __esModule: true,
  MainNav: jest.fn(),
}));

jest.mock('@/components/theme-picker', () => ({
  __esModule: true,
  default: jest.fn(),
}));

jest.mock('@/components/user/user-nav', () => ({
  __esModule: true,
  UserNav: jest.fn(),
}));

jest.mock('@/components/ui/toaster', () => ({
  __esModule: true,
  Toaster: jest.fn(),
}));

// Set up mock implementation for window.matchMedia
window.matchMedia = jest.fn().mockImplementation(query => {
  return {
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
    dispatchEvent: jest.fn(),
  };
});


describe('Dashboard layout component', () => {

  beforeEach(() => {
    LogoImage.mockImplementation(() => <div>Ethereal Nexus Logo</div>);
    MainNav.mockImplementation(() => <div>Main Nav</div>);
    ThemePicker.mockImplementation(() => <div>ThemePicker</div>);
    UserNav.mockImplementation(() => <div>User Nav</div>);
    Toaster.mockImplementation(() => <div>Toaster</div>);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    const { asFragment } = render(
      <DashboardLayout>
        <div>Child Component</div>
      </DashboardLayout>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render all child components', () => {
    render(
      <DashboardLayout>
        <div>Child Component</div>
      </DashboardLayout>
    );

    expect(screen.getByText('Ethereal Nexus Logo')).toBeInTheDocument();
    expect(screen.getByText('Main Nav')).toBeInTheDocument();
    expect(screen.getByText('ThemePicker')).toBeInTheDocument();
    expect(screen.getByText('User Nav')).toBeInTheDocument();
    expect(screen.getByText('Toaster')).toBeInTheDocument();
  });

  it('should render children within the layout', () => {
    render(<DashboardLayout>Test Content</DashboardLayout>);

    // Test that children are rendered
    expect(screen.getByText('Test Content')).toBeInTheDocument();
  });
});
