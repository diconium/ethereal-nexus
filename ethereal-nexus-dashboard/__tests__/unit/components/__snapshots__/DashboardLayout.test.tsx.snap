// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard layout component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex-col md:flex relative mt-6"
  >
    <div
      class="relative"
    >
      <div
        class="flex h-16 items-center px-8"
      >
        <div>
          Ethereal Nexus Logo
        </div>
        <div>
          Main Nav
        </div>
        <div
          class="ml-auto flex items-center space-x-4"
        >
          <div>
            ThemePicker
          </div>
          <div>
            User Nav
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex-1 space-y-4 p-8 mt-6 absolute"
    >
      <div>
        Child Component
      </div>
    </div>
    <div>
      Toaster
    </div>
  </div>
</DocumentFragment>
`;
